rest_docs/rest-catalog-open-api-v0.1.yaml

```diff
@@ -0,0 +1,657 @@
+        in: path
+        required: true
+        schema:
+          type: string
+        examples:
+          singlepart_namespace:
+            value: "accounting"
+          multipart_namespace:
+            value: "accounting.tax"
+    get:
+      tags:
+        - Catalog API
+      summary: List all table identifiers underneath a given namespace
+      description: Return all table identifiers under this namespace
+      operationId: listTables
+      responses:
+        '200':
+          description: OK
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ListTablesResponse'
+        '404':
+          description: Not Found (NoSuchNamespaceException)
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/responses/IcebergResponseObject'
+              example:
+                '{ error: { message: "Namespace does not exist", type: "NoSuchNamespaceException", code: 404 }'
+
+  /namespaces/{namespace}/tables/{table}:
+    parameters:
+      - name: namespace
+        in: path
+        description: A namespace identifier
+        required: true
+        schema:
+          type: string
+        examples:
+          singlepart_namespace:
+            value: "accounting"
+          multipart_namespace:
+            value: "accounting.tax"
+      - name: table
+        in: path
+        description: A table name
+        required: true
+        schema:
+          type: string
+        example: "sales"
+
+    delete:
+      tags:
+        - Catalog API
+      summary: Drop a table from the catalog
+      operationId: dropTable
+      description: Remove a table from the catalog
+      parameters:
+        - name: purgeRequested
+          in: query
+          required: false
+          description: Whether the user requested to purge the underlying table's data and metadata
+          schema:
+            type: boolean
+            default: false
+      responses:
+        200:
+          description: OK
+          content:
+            'application/json':
+              schema:
+                $ref: '#/components/schemas/DropTableResponse'
+              example: { "data": { "dropped": true, "purged": false } }
+        202:
+          # TODO - Add a callback or handle to check on the state of an async purge request
+          description: Accepted - for use if purgeRequested is implemented as an asynchronous action.
+        404:
+          $ref: '#/components/responses/NoSuchTableResponse'
+
+    head:
+      tags:
+        - Catalog API
+      summary: Check if a table exists
+      operationId: tableExists
+      description:
+        Check if a table exists within a given namespace.
+      responses:
+        200:
+          description: OK - Table Exists
+        401:
+          description: Unauthorized
+        404:
+          description: Not Found
+
+  /tables/rename:
```

Comment:
I am concern here to mix to functionalities into one method. Why don't we just separate these two functionalities and move under the table endpoint? E.g:
`PUT /namespaces/{namespace}/tables/{table}/rename`
`PUT /namespaces/{namespace}/tables/{table}/move`

Reply:
Oh. So using `PUT` with `/rename` might not be considered as overloading.

That's a good suggestion.

As for with `rename`vs`move`, can you clarify further which two functionalities you mean? What would `rename` do that `move` wouldn't, and similarly what would `move` do that `rename` wouldn't? Or when would each be used.

Reply:
But with your proposed `/rename` and then `/move` route, it would be supportable once that exists.

Reply:
My point is, from the description of this REST endpoint, this endpoint does two things, rename a table and move a table to new namespace. My problem with that, I find it a bit weird to include several actions in one REST request which could be misleading to the end user IMHO.  Therefore, I was proposing if we could maybe break isolate these two functionalities and keep each by its own REST endpoint, also using what I proposed earlier and add a scope argument, we can limit this to move across namespaces, e.g: 
```
PUT /namespaces/{namespace}/tables/{table}/actions?action=move&scope=namespace
PUT /namespaces/{namespace}/tables/{table}/actions?action=rename
```
This is just an idea to have for example a scope argument to only execute this action within that scope. So in the future, when we have move functionality for data, then the scope could be `?scope=data` for example. makes sense?

By the way, I think what I proposed earlier to have verbs in the REST URL may not go well with the REST guidelines, thus perhaps `actions` noun and the verb as argument would make more sense?

Reply:
My concern with this is that what you're calling `move` and `rename` are currently just one action. So i think it would be possibly misleading to have them as two separate things, when they are really one thing (at least in the current API).

The rename table doesn't move any of the data presently. So what would happen if somebody `PUT ...action=move` and that worked, and then `PUT ....action=rename` and that call failed? What state would the system be in? Or what state would the system be in even after just a `move` call?

As for the `/actions` idea, I personally do like that and wouldn't be opposed.

I'll keep it as is for now so this discussion doesn't get moved / erased, and then we can talk about the `/actions` thing coming up.

This is still very much a work in progress. So here would sitll be time for later changes.